erpc_c/transports/erpc_uart_espidf_transport.cpp

```diff
@@ -85,20 +85,13 @@ erpc_status_t UartTransport::init(int timeout)
 
 erpc_status_t UartTransport::underlyingReceive(uint8_t *data, uint32_t size)
 {
-    int len = -1;
-
-    TickType_t timeout = (size * 10 * 1000 * 2) / m_uart.config.baud_rate / portTICK_RATE_MS;
-
-    while (size &&
-           ((len = uart_read_bytes(
-                 m_uart.num, data, size, timeout < m_timeout ? m_timeout : timeout)) > 0))
-    {
-        size -= len;
-    }
-
-    if (size)
-    {
-        return (len == -1) ? kErpcStatus_ReceiveFailed : kErpcStatus_Timeout;
+    while (size != 0) {
+        int read_res = uart_read_bytes(m_uart.num, data, size, portMAX_DELAY);
+        if (read_res == -1) {
```

Comment:
Is -1 same as ESP_FAIL? Can we use ESP_FAIL instead?

Reply:
The documentation at https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html#_CPPv415uart_read_bytes11uart_port_tPv8uint32_t10TickType_t says it returns "(-1) Error", so I'd rather see the comparison here against -1.